plumbing/protocol/packp/srvresp.go

```diff
@@ -52,31 +80,71 @@ func (r *ServerResponse) decodeLine(line []byte) error {
 		}
 
 		if bytes.Equal(line[0:3], nak) {
-			return nil
+			return io.EOF
 		}
 	}
 
 	return fmt.Errorf("unexpected content %q", string(line))
 }
 
-func (r *ServerResponse) decodeACKLine(line []byte) error {
-	if len(line) < ackLineLen {
+func (r *ServerResponse) decodeACKLine(line []byte) (err error) {
+	parts := bytes.Split(line, []byte(" "))
+	if len(line) < ackLineLen || len(parts) < 2 {
 		return fmt.Errorf("malformed ACK %q", line)
 	}
 
-	sp := bytes.Index(line, []byte(" "))
-	h := plumbing.NewHash(string(line[sp+1 : sp+41]))
-	r.ACKs = append(r.ACKs, h)
-	return nil
+	var ack ACK
+	// TODO: Dynamic hash size and sha256 support
```

Comment:
Again, let's pick up on a future PR, but for packfile this is based on an "option" as the packfile can do SHA1 or SHA1 and SHA256 hashes. Here, I believe it would need to be either/or and would be based off the object format.

Reply:
Right, we will need to support the `object-format` capability. I think here, instead of hard-coding the hash lengths, we need to be more lenient and just treat whatever argument as a hash.